src/dodal/devices/oav/snapshots/snapshot_with_grid.py

```diff
@@ -0,0 +1,64 @@
+from io import BytesIO
+from os.path import join as path_join
+
+import aiofiles
+from ophyd_async.core import soft_signal_rw
+from PIL.Image import Image
+
+from dodal.devices.areadetector.plugins.MJPG import MJPG
+from dodal.devices.oav.snapshots.grid_overlay import (
+    add_grid_border_overlay_to_image,
+    add_grid_overlay_to_image,
+)
+from dodal.log import LOGGER
+
+
+class SnapshotWithGrid(MJPG):
+    def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(prefix, name)
```

Comment:
Should: I think you need the `super().__init__` after the signal definition so that the names of the signals are correctly filled in